main_2024.py

```diff
@@ -255,6 +276,130 @@ def main() -> int:
                 print("geolocation confidence: " + str(detection_world.confidence))
                 print("")
 
+        # Check if all workers are alive in each manager.
+        # If a worker is dead, log the error and restart the worker.
+        # Steps: Terminate and join the worker, fill and drain the
+        # related queues, and create and start a new worker.
+        for manager in range(number_of_managers):
+            if not managers_array[manager].are_workers_alive():
+                managers_array[manager].terminate_workers()
+                managers_array[manager].join_workers()
+                managers_array[manager] = None
+
+                if manager == ManagerType.VIDEO_INPUT.value:
+                    main_logger.error(
+                        "Video Input Worker is dead, attempting to restart.",
+                        frame,
+                    )
+                    video_input_to_detect_target_queue.fill_and_drain_queue()
+                    del video_input_manager
+                    video_input_manager = worker_manager.WorkerManager()
+                    managers_array[manager] = video_input_manager
+                    video_input_manager.create_workers(
+                        1,
+                        video_input_worker.video_input_worker,
+                        (
+                            VIDEO_INPUT_CAMERA_NAME,
+                            VIDEO_INPUT_WORKER_PERIOD,
+                            VIDEO_INPUT_SAVE_PREFIX,
+                            video_input_to_detect_target_queue,
+                            controller,
+                        ),
+                    )
```

Comment:
we should change the rest of main to use a variable for the args so it's reusable (and so we can pass the args into a function), so we would have a variable named video_input_worker_args, etc.